### DISCUSSION TITLE
Multiapp and changing subdomain ID

### DISCUSSION THREAD

**Discussion by Wendy-Ji** [idaholab/moose#28861] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello,

I am getting a segmentation fault for a particular case of changing the subdomain ID of elements in a multiapp.

The example I put below has two subdomains (1 and 2), and an empty subdomain (3). The elements in subdomain 2 change to subdomain 3, and the variable $\phi$ is defined over subdomains 1 and 2.

When I run just this scenario, i.e. the subapp code, the simulation runs fine.

When I run the same scenario in a multiapp, i.e. the master and subapp code below, with the variable $\phi$ transferring from the subapp to the master, then a segmentation fault occurs in the master app when the elements change subdomain ID. 

This multi-app setup works fine for other cases of changing subdomain ID, so I'm not sure what's wrong here.

Subapp:
```
[Problem]
  solve = false
[]

[Mesh]
  [gen]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 2
    ny = 2
  []
  [left]
    type = SubdomainBoundingBoxGenerator
    input = 'gen'
    block_id = 1
    bottom_left = '0 0 0'
    top_right = '0.5 1 1'
  []
  [right]
    type = SubdomainBoundingBoxGenerator
    input = 'left'
    block_id = 2
    bottom_left = '0.5 0 0'
    top_right = '1 1 1'
  []
  add_subdomain_ids = 3
[]

[Variables]
  [phi]
    block = '1 2'
  []
[]

[MeshModifiers]
  [deact]
    type = TimedSubdomainModifier
    times = '1'
    blocks_from = '2'
    blocks_to = '3'
    execute_on = 'INITIAL TIMESTEP_BEGIN'
  []
[]

[Executioner]
  type = Transient
  dt = 0.5
  num_steps = 3
[]
```
Master app:
```
[Problem]
  solve = false
[]

[Mesh]
  [gen]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 2
    ny = 2
  []
  [left]
    type = SubdomainBoundingBoxGenerator
    input = 'gen'
    block_id = 1
    bottom_left = '0 0 0'
    top_right = '0.5 1 1'
  []
  [right]
    type = SubdomainBoundingBoxGenerator
    input = 'left'
    block_id = 2
    bottom_left = '0.5 0 0'
    top_right = '1 1 1'
  []
  add_subdomain_ids = 3
[]

[AuxVariables]
  [phi_aux]
    block = '1 2'
  []
[]

[MeshModifiers]
  [deact]
    type = TimedSubdomainModifier
    times = '1'
    blocks_from = '2'
    blocks_to = '3'
    execute_on = 'INITIAL TIMESTEP_BEGIN'
  []
[]

[Executioner]
  type = Transient
  dt = 0.5
  num_steps = 3
[]

[MultiApps]
  [multiapp]
    type = TransientMultiApp
    positions = '0.0 0.0 0.0'
    input_files = 'subapp.i'
  []
[]

[Transfers]
  [phi_from_subapp]
    type = MultiAppCopyTransfer
    from_multi_app = 'multiapp'
    source_variable = 'phi'
    variable = 'phi_aux'
  []
[]
```

**Comment by GiudGiud** [comment, marked as answer]
Hello

The `MultiAppCopyTransfer` is meant for meshes that are exactly the same. Since you are changing the mesh, they are no longer the same. Please use this one instead:
https://mooseframework.inl.gov/source/transfers/MultiAppGeneralFieldShapeEvaluationTransfer.html
